lib/value/src/kind/collection.rs

```diff
@@ -107,18 +105,32 @@ impl<T: Ord> Collection<T> {
         &mut self.known
     }
 
-    /// Get a reference to the "unknown" elements in the collection.
-    ///
-    /// If `None` is returned, it means all elements within the collection are known, i.e. it's
-    /// a "closed" collection.
+    /// Gets the type of "unknown" elements in the collection.
+    /// The returned type will always have "undefined" included.
```

Comment:
I'll add a comment, but today either `never` or `undefined` would be correct. I'd like the "canonicalized" form to eventually be just "undefined" since that's the _actual_ type of types that aren't defined with a closed type.